scripts/Diffraction/wish/wishSX.py

```diff
@@ -0,0 +1,129 @@
+from typing import Sequence
+import numpy as np
+import mantid.simpleapi as mantid
+from Diffraction.single_crystal.base_sx import BaseSX
+from mantid.api import AnalysisDataService as ADS
+
+
+class WishSX(BaseSX):
+    def __init__(self, vanadium_runno=None):
+        super().__init__(vanadium_runno)
+        self.grp_ws = None  # banks grouping workspace
+        self.ngrp = None  # no. groups (i.e. number of banks)
+        self.sphere_shape = """<sphere id="sphere">
+                               <centre x="0.0"  y="0.0" z="0.0" />
+                               <radius val="0.0025"/>
+                               </sphere>"""  # sphere radius 2.5mm  - used for vanadium and NaCl
+
+    def process_data(self, runs: Sequence[str], *args):
+        gonio_angles = args
+        if len(gonio_angles) != len(self.gonio_axes):
+            return
+        for irun, run in enumerate(runs):
+            wsname = self.load_run(run)
+            # set goniometer
+            if self.gonio_axes is not None:
+                # gonio_angles are a list of motor strings (same for all runs)
+                if isinstance(gonio_angles[0], str):
+                    self._set_goniometer_on_ws(wsname, gonio_angles)
+                else:
+                    # gonio_angles is a list of individual or tuple motor angles for each run
+                    self._set_goniometer_on_ws(wsname, gonio_angles[irun])
+            # correct for empty counts and normalise by vanadium
+            self._divide_workspaces(wsname, self.van_ws)
+            # set sample (must be done after gonio to rotate shape) and correct for attenuation
+            if self.sample_dict is not None:
+                mantid.SetSample(wsname, EnableLogging=False, **self.sample_dict)
+                mantid.ConvertUnits(InputWorkspace=wsname, OutputWorkspace=wsname, Target="Wavelength", EnableLogging=False)
```

Comment:
Similar comment to the one in the sxd.py unit - are the units on the ws at the end of this function different depending on whether the absorption correction is done. Think the final units are wavelength if code comes in here